flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/AbstractHaServices.java

```diff
@@ -225,6 +237,13 @@ public CompletableFuture<Void> globalCleanupAsync(JobID jobID, Executor executor
                 executor);
     }
 
+    protected void cleanupClusterHaStoragePath() throws Exception {
+        final Path clusterHaStoragePath =
+                HighAvailabilityServicesUtils.getClusterHighAvailableStoragePath(configuration);
+        final FileSystem fileSystem = clusterHaStoragePath.getFileSystem();
+        fileSystem.delete(clusterHaStoragePath, true);
```

Comment:
@dmvk noticed that this might break the JobResultStore if [job-result-store.delete-on-commit](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#job-result-store-delete-on-commit) is disabled but the default JobResultStore directory is not overridden (see [job-result-store.storage-path](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#job-result-store-storage-path)). The default path relies on [high-availability.storageDir](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#high-availability-storagedir) which we use here to delete all artifacts in this directory.